File: packages/lynx_view_android/android/src/main/kotlin/com/geektree0101/lynx_view_android/LynxPlatformViewFactory.kt
package com.geektree0101.lynx_view_android

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

/** viewType this factory answers to — must match the Dart-side platform interface. */
internal const val LYNX_VIEW_TYPE = "com.geektree0101.lynx_view/LynxView"

internal class LynxPlatformViewFactory(
    private val binaryMessenger: BinaryMessenger,
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        LynxViewPlugin.ensureInitialized(context)
        @Suppress("UNCHECKED_CAST")
        val creationParams = args as? Map<String?, Any?>
        // Before the view exists, not after: Lynx reads its typeface cache
        // while measuring text, and a font registered late shows up as a
        // reflow (or not at all, if the template never re-measures).
        (creationParams?.get("fonts") as? List<*>)?.let { FontAssets.register(context, it) }
        return LynxPlatformView(context, viewId, binaryMessenger, creationParams)
    }
}

File: packages/lynx_view_ios/ios/Classes/LynxPlatformViewFactory.swift
import Flutter
import Foundation

/// viewType this factory answers to — must match the Dart-side platform interface.
let lynxViewType = "com.geektree0101.lynx_view/LynxView"

final class LynxPlatformViewFactory: NSObject, FlutterPlatformViewFactory {
    private let messenger: FlutterBinaryMessenger

    init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
        super.init()
    }

    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        LynxViewPlugin.ensureInitialized()
        // Before the view exists, not after: Lynx resolves the typeface while
        // measuring text, and a font registered late shows up as a reflow (or
        // not at all, if the template never re-measures).
        if let params = args as? [String: Any], let fonts = params["fonts"] as? [Any] {
            FontAssets.register(fonts)
        }
        return LynxPlatformView(frame: frame, viewId: viewId, arguments: args, binaryMessenger: messenger)
    }

    func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        FlutterStandardMessageCodec.sharedInstance()
    }
}
